change_date_rename_file.sh

```diff
@@ -80,7 +80,22 @@ while read a_file_name; do
   # Replace IMG in filename with value in $datestring_for_filename, which is
   # in the format: YYYY-MM-DD_HH-MM, e.g., 2016-01-27_08-15.
   new_file_name="${a_file_name/IMG/$date_for_filename_change}"
-  mv "$a_file_name" "$new_file_name"
+ 
+  # Build onto new_file_name. Concat to front - year & month.
+  # Test with: /Users/kimlew/Sites/bash_projects/test_mkdir-p
+  # Use: ${string:position:length}   On: 2015:09:02 07:09:03
+  year=${exif_date:0:4}
+  month=${exif_date:5:2}
+
+  just_path=$(dirname "${new_file_name}")
+  path_with_subdir_year_month="${just_path}/${year}/${month}"
+
+  mkdir -p ${path_with_subdir_year_month}
+
+  just_filename=$(basename "${new_file_name}")
+  new_dir_and_filename=$just_path/$year/$month/$just_filename
```

Comment:
In your other parameter expansions, like `${just_path}/${year}/${month}` above, you're using curly braces around the variable names, but here you don't. It's not wrong, but you should use a consistent style and add the curly braces here.

Reply:
@kcharter As you suggested, I made the change so it is more consistent with other brace expansions in script.